hack/packages/package-tools/cmd/repo-bundle-generate.go

```diff
@@ -0,0 +1,285 @@
+	if err := utils.CreateDir(filepath.Join(projectRootDir, constants.RepoBundlesDir, packageRepository, ".imgpkg")); err != nil {
+		return err
+	}
+
+	if err := utils.CreateDir(filepath.Join(projectRootDir, constants.RepoBundlesDir, packageRepository, "packages")); err != nil {
+		return err
+	}
+
+	toolsBinDir := filepath.Join(projectRootDir, constants.ToolsBinDirPath)
+
+	// generate repo bundle image lock output file
+	yttCmd := exec.Command(filepath.Join(toolsBinDir, "ytt"), "-f", filepath.Join(projectRootDir, "hack", "packages", "templates", "repo-utils", "images-tmpl.yaml"),
+		"-f", filepath.Join(projectRootDir, "hack", "packages", "templates", "repo-utils", "package-helpers.lib.yaml"),
+		"-f", packageValuesFile,
+		"-v", "packageRepository="+packageRepository,
+		"-v", "registry="+registry) // #nosec G204
+
+	outFilePath := filepath.Join(projectRootDir, constants.RepoBundlesDir, packageRepository, ".imgpkg", "images.yml")
+	outfile, err := os.Create(outFilePath)
+	if err != nil {
+		return fmt.Errorf("error creating file %s : %w", outFilePath, err)
+	}
+	defer outfile.Close()
+	var errBytes bytes.Buffer
+	yttCmd.Stdout = outfile
+	yttCmd.Stderr = &errBytes
+
+	if err = yttCmd.Run(); err != nil {
+		return fmt.Errorf("couldn't generate the image lock output file: %s", errBytes.String())
+	}
+
+	packageValuesData, err := os.ReadFile(packageValuesFile)
+	if err != nil {
+		return fmt.Errorf("couldn't read file %s: %w", packageValuesFile, err)
+	}
+
+	packageValues := PackageValues{}
+	if err := yaml.Unmarshal(packageValuesData, &packageValues); err != nil {
+		return fmt.Errorf("error while unmarshaling: %w", err)
+	}
+
+	repository, found := packageValues.Repositories[packageRepository]
+	if !found {
+		return fmt.Errorf("%s repository not found", packageRepository)
+	}
+
+	for i := range repository.Packages {
+		if err := generatePackage(projectRootDir, toolsBinDir, &repository.Packages[i]); err != nil {
+			return fmt.Errorf("couldn't generate the package: %w", err)
+		}
+	}
+
+	// create tarball of repo bundle
+	tarBallPath := filepath.Join(projectRootDir, constants.RepoBundlesDir)
+	tarBallFileName := "tanzu-framework-" + packageRepository + "-repo-" + version + ".tar.gz"
+	pathToContents := filepath.Join(tarBallPath, packageRepository)
+	if err := utils.CreateTarball(tarBallPath, tarBallFileName, pathToContents); err != nil {
+		return fmt.Errorf("couldn't generate package bundle: %w", err)
+	}
+
+	// move the tar ball to repository directory
+	if err := os.Rename(filepath.Join(tarBallPath, tarBallFileName), filepath.Join(tarBallPath, packageRepository, tarBallFileName)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func generatePackage(projectRootDir, toolsBinDir string, pkg *Package) error {
+	if err := utils.CreateDir(filepath.Join(projectRootDir, constants.RepoBundlesDir, packageRepository, "packages", pkg.Name+"."+pkg.Domain)); err != nil {
+		return err
+	}
+
+	pkgVersion := version[1:]
```

Comment:
It seems that the version used for the package filenames gets trimmed incorrectly. For example, if running with `REPO_BUNDLE_VERSION=1.2.3` the package yamls generated end up as `.2.3.yml`.

We should either not trim the first character here or check that the first character is something we want to remove.

Reply:
@adobley thanks for catching this, i made a fix and pushed the code..